architecture-decision-records/0022-cloudfront-upload-proxy.md

```diff
@@ -0,0 +1,164 @@
+# 22. Proxy S3 uploads through CloudFront
+
+**Date:** 2020-09-17
+
+## Context
+
+When a user uploads a file to TDR, the file is currently sent from the JavaScript running in the user's browser straight
+to the AWS S3 upload bucket. This means that there are several HTTP requests from the user's browser to AWS domains:
+
+- `cognito-identity.eu-west-2.amazonaws.com`: Exchange a TDR access token for a Cognito token
+- `sts.amazonaws.com`: Exchange the Cognito token for a long-lived STS token
+- `tdr-upload-files-dirty-<environment>.s3.eu-west-2.amazonaws.com`: Upload the files
+
+The sequence of requests looks like this (simplified slightly - CORS requests and multiple requests to the auth server
+and S3 are omitted for clarity):
+
+![Sequence diagram showing current upload process](images/upload-sequence-diagram-cognito.png)
+
+Some TDR users are on restricted networks, and the upload step completely fails. From what we could see on a video call
+with one user, the network calls to TDR domains are succeeding but the ones to `amazonaws.com` are being blocked.
```

Comment:
Maybe reword slightly to: "... calls to TDR domains succeed but the ones to ..."

Reply:
Done.